File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/builder/ConcreteBuilder.java
package builder;

public class ConcreteBuilder implements Builder {
    private Product product;

    public ConcreteBuilder() {
        this.product = new Product();
    }

    public void buildPartA() {
        product.addPart("Whole Wheat Bread");
    }

    public void buildPartB() {
        product.addPart("Turkey");
    }

    public void buildPartC() {
        product.addPart("Lettuce, Tomato, Onion");
    }

    public Product getProduct() {
        return product;
    }
}

File: code/csharp/DesignPatterns/builder/ConcreteBuilder.cs
namespace Builder;

class ConcreteBuilder : IBuilder
{
    private Product _product = new Product();

    public void BuildPartA()
    {
        _product.AddPart("Whole Wheat Bread");
    }

    public void BuildPartB()
    {
        _product.AddPart("Turkey");
    }

    public void BuildPartC()
    {
        _product.AddPart("Lettuce, Tomato, Onion");
    }

    public Product GetProduct()
    {
        return _product;
    }
}
